docs/project-development.md

```diff
@@ -1,12 +1,94 @@
 Project Development
 ---
 
-A common structure for Reflex-DOM applications is to have a `common`
-package shared by a `backend` and a `frontend`. To accomplish this
-with `reflex-platform`, create the three cabal packages, and then make
-a project file called `default.nix`.
+This document describes how to build real-world applications written
+in Reflex. You will see how to:
+
+- [Create a project from scratch](#creating-the-project)
+- [Build it with Nix](#building-with-nix)
+- [Develop it incrementally](#building-with-cabal)
+- [Test it quickly](#building-frontends-with-ghc)
+- [Make simple mobile apps](#building-mobile-apps)
+
+Creating the Project
+---
+
+First, create a directory for your project. This will contain all of
+the files needed for the build process and a checkout of
+`reflex-platform`, which will provide all of the Haskell libraries and
+compilers the project will depend on. To get `reflex-platform`, it is
+easiest to use [git](https://git-scm.com/) and add it as a submodule,
+so that the version being used is consistent amongst your team and
+updating it is easy.
+
+```bash
+$ mkdir my-project
+$ cd my-project
+$ git init
+$ git submodule add https://github.com/reflex-frp/reflex-platform
+```
+
+If you've never built a project with `reflex-platform` before, you may
+need to install [Nix](https://nixos.org/nix/) and configure Reflex's
+binary cache. `reflex-platform` provides the `try-reflex` script,
+which will do this for you and download some of the basic tools and
+libraries we'll need ahead of time.
+
+```bash
+$ reflex-platform/try-reflex
+```
+
+After running this command, you'll find yourself in a different
+shell. This is the `try-reflex` sandbox, which provides GHC and GHCJS
+with `reflex-dom` preinstalled. You can use this environment to
+quickly test things out, but this document only uses it to install
+Nix, so go ahead and `exit` out of this shell.
```

Comment:
If we want to simplify this explanation, we can write a tiny script that just installs nix and adds our binary caches.